gammapy/scripts/config/schema.yaml

```diff
@@ -44,47 +44,161 @@ properties:
     observations:
         description: "Observations used in the analysis."
         type: object
+        default: {}
         additionalProperties: false
         properties:
-            data_store:
-                description: "Data store where to fetch the observations."
+            datastore:
+                description: "Data store where to fetch observations."
                 type: string
+                default: $GAMMAPY_DATA/hess-dl3-dr1/hess-dl3-dr3-with-background.fits.gz
             filter:
+                description: "Array of filter criteria to select observations."
                 type: array
-                items: {"$ref": "#/definitions/observations_filter"}
+                default: [{}]
+                items:
+                    type: object
+                    additionalProperties: false
+                    properties:
+                        exclude: {type: boolean}
+                        inverted: {type: boolean}
+                        filter_type:
+                            default: par_value
+                            enum: [sky_circle,
+                                   angle_box,
+                                   par_box,
+                                   par_value,
+                                   ids,
+                                   all]
+                        frame:
+                            default: icrs
+                            enum: [galactic, equatorial, icrs, fk5]
+                        lon: {type: number, default: 83.633 deg}
```

Comment:
If the default is "83.633 deg" the type has to be string or quantity, no?
You write `type: number`.
!?

There's a few config items where probably we don't want to offer a default, target position being one of them.

Otherwise people will forget to change it, and run a Crab position analysis, which will not be what they wanted in 99% of the cases.

Is it possible to have config items without a default in our config scheme?
Can you make the target position required?

Reply:
*re: type: number*
we have our own validator that extends the one of `jsonschema` and re-defines validation conditions (adding support for astropy quantities) to one type among a set of pre-defined types. I chose number instead of string because it was easier and also because quantities have numbers so we inherit the built-in validation for numbers. This is only unsed internally for validation, users do not have to know this subtleties in the jsonschema syntax of `schema.yaml` file, they don't have to provide types nor worrying about if what they provide is a number or a string, as long as it may be converted to an astropy quantity.

Yes, it is possible to have config items without a default in our config scheme. It's up to us to decide what settings have a default, which ones are required and under which conditions. (i.e. the value of a setting may change the default of another or require/disable the presence of others, etc.) 

I am using the `schema.yaml` to have default values for the specific use case I'm facing, we will provide a clean  `schema.yaml`  with good default generic values at the end. FOr the moment, just consider the overall structure of the file, the names of the properties and where they are placed, not the specific values.

Reply:
Would be good to have an example very soon of config items without defaults. There won't be many, and we won't touch them often, because every time you have to adjust most tests and docs examples. Would start with position.

Reply:
Ok, as it was discussed in last dev call, this will be done in another PR.